spec/support/rspec_example_trace_helpers.rb

```diff
@@ -19,15 +19,32 @@ def pending_fixed?
 end
 
 module RSpecExampleTraceHelpers
-  def fake_example(status:)
-    example = double("RSpec::Core::Example")
```

Comment:
The motivation for changing this helper was due to adding some additional tests on `file_path` and `location`.

The RSpec library hook looks at the source location of the `example`, in this case this is the declaration  location of the RSpec `double` method, which is usually buried somewhere obscure like `/home/user/path/to/ruby/version/manager/gems/rspec-1.2.3/lib/rspec/double.rb` or some such weird path, different in every build environment. This makes testing the correct path and location a little difficult.

I changed this helper to return a verifying `instance_double`, and additionally allowed injection of the `file_path` and `location` from the test possible to make testing these properties easier.